ElementX/Sources/Services/Timeline/TimelineItemProxy.swift

```diff
@@ -120,10 +120,46 @@ class EventTimelineItemProxy {
         let debugInfo = item.debugInfo()
         return TimelineItemDebugInfo(model: debugInfo.model, originalJSON: debugInfo.originalJson, latestEditJSON: debugInfo.latestEditJson)
     }()
-
+    
+    lazy var shield: MessageShield? = {
+        if let shield = item.getShield(strict: false) {
+            return MessageShield.fromRustShield(shieldState: shield)
+        }
+        return nil
+    }()
+    
     lazy var readReceipts = item.readReceipts()
 }
 
+extension MessageShield {
+    static func fromRustShield(shieldState: ShieldState) -> MessageShield? {
```

Comment:
This is really an init so may as well use once :)
```suggestion
    init(rustShieldState: ShieldState) {
```

Reply:
It may not create a MessageShield

Reply:
My bad, we can use `init?` for optional initialisers in Swift, have updated the suggestion.